### PR TITLE
Add hive repo to CI.

### PR DESCRIPTION
**Pull request by twiest**
- [x] Add Added prow configs
- [x] Add ci-operator configs.
- [x] Generate ci-operator jobs using ci-operator-prowgen.
- [x] Test ci-operator configs using local ci-operator and oc cluster up.

### CODE REVIEW

File: ci-operator/config/openshift/hive/master.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      cpu: '2'
+      memory: 1Gi
+tag_specification:
+  cluster: https://api.ci.openshift.org
+  name: origin-v4.0
+  namespace: openshift
+  tag: ''
+  tag_overrides: {}
+build_root:
+  image_stream_tag:
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/hive/master.yaml]: I think what you want is to use `project_image` here, not the tag, and set the project build to use your builder-base.

Anything like:
```
# Create the full dir tree that we'll mount our src into when we run the image
RUN mkdir -p /go/src/github.com/openshift/hive

# Default to our src dir
WORKDIR /go/src/github.com/openshift/hive
```
is an implementation detail of the CI and you should not need to do it at all

you should not need your test-base image

File: ci-operator/config/openshift/hive/master.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+tests:
+- as: unit
+  commands: make test
+  from: test-base
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/hive/master.yaml]: So you should be able to get away with:

```yaml
build_root:
  project_image:
    dockerfile_path: build/test-image/Dockerfile
tests:
- as: unit
  commands: make test
  from: src
```

and drop the `RUN mkdir -p` and the `WORKDIR` from the builder-base, drop the test-base entirely

Comment (twiest) [reply ci-operator/config/openshift/hive/master.yaml]: Hey Steve, thanks for the review.

When I make the changes you suggest and run ci-operator locally (0771d83 on master branch), I get this error:
```
[twiest@vm hive (add-hive-to-ci *%=)]$ ci-operator --config  $HOME/code/release/ci-operator/config/openshift/hive/master.yaml --target unit --git-ref "twiest/hive@ci-changes"
error: invalid configuration: you have to specify either git_source_image or image_stream_tag for the build_root
```

I don't see git_source_image in the [ci-operator documentation](https://github.com/openshift/ci-operator/blob/master/CONFIGURATION.md#build_root), so I'm not sure how to proceed.

Here's my `build_root` and `tests` sections:
```yaml
build_root:
  project_image:
    dockerfile_path: build/build-image/Dockerfile
tests:
- as: unit
  commands: make test
  from: src
```

Comment (stevekuznetsov) [reply ci-operator/config/openshift/hive/master.yaml]: Oops, we had an issue with that config validation check. I've fixed that, can you try again with a fresh pull of the `ci-operator` image or a re-build from `master`?

**Pull request merged**